### DISCUSSION TITLE
Authentication layer for pygeoapi?

### DISCUSSION THREAD

**Discussion by tjukanovt** [geopython/pygeoapi#833] [General]
Howdy! 

I am currently looking into pygeoapi and it seems like a very nice piece of software. So firstly thanks to all the contributors. 

Then my question: if you were to set up a pygeoapi in a production environment and publish different services through it, what are my options on handling authentication? So for example if I would like to have an API key authentication for users to use the processes API, what are my options in building something like that? I assume this would have to be done with a separate piece of software as there isn't anything on this topic in the docs. 

Sorry if this is an irrelevant question. Just didn't find anything on the topic with quick googling and thought this might be an interesting topic also for others to discuss.

**Comment by francbartoli** [comment]
Hi @tjukanovt, there isn't such an authentication layer at the moment. Would you like to file a ticket for that? I think it will be a useful feature

**Reply by francbartoli** [reply]
Issue opened https://github.com/geopython/pygeoapi/issues/867

cc'ed @tomkralidis @kalxas @staffordsmith83

**Comment by francbartoli** [comment]
Authentication and authorization layers are going to be delegated to implementers in external plugins and downstream projects. This will avoid to do not force developers to use only supported security mechanisms which have been hacked in the web frameworks of pygeoapi. It will save flexibility and modularity in the long term

**Comment by ridoo** [comment]
You could have a look at 

  https://github.com/geobeyond/fastgeoapi

which provides you with components (incl. pygeoapi) leveraging authentication and authorization (via OPA policies).
